docs/rfcs/0004-suggested-excluded-apps-tech-spec.md

```diff
@@ -0,0 +1,142 @@
+
+Context / Problem Statement
+---------------------------
+
+We're looking to create tools that prompt the user in ways that help them stay more secure. Split tunneling (called 'App Exclusion' to users) is a feature that can reduce the need to turn off the VPN entirely by excluding traffic in specific apps from the VPN tunnel, but many users don't know about it. The current feature is in settings, and provides an alphabetical checklist of installed apps:
+
+<img src="./images/0004-01.png" height=500>
+
+This epic is for a new feature to proactively let users know when they have apps that could benefit from the split tunneling feature. We want this to be a notification outside the app, to engage users who rely on Mozilla VPN but rarely open the app.
+
+Assumptions & Out-of-scope
+--------------------------
+
+-   This is only for platforms that currently support split tunnel - Windows, Linux, and Android. We will not build split tunneling into macOS or iOS as part of this work.
+-   Notifications refer to system-level notifications. There are no additions to the messages tab considered in this tech spec.
+
+Open Questions
+--------------
+
+-   What do we use to canonically track apps we may want to exclude? Is there a bundle ID? The name of app (fragile)? Something else? Do we use different IDs across platforms?
+-   How hard would these checks be to run in Android daemon or PeriodicWorkRequest? (If substantially harder, Santiago open to doing it in Android app - which reduces how often it runs, as the app must be open in order for it to run.) On desktop platforms, the app is always active when the VPN is active. This is not true on mobile, which is why this is considered separately.
+-   Any easy ways to categorize games? (See [VPN-4433](https://mozilla-hub.atlassian.net/browse/VPN-4433).)
+-   How will we decide which apps are initially on the suggested list?
+-   Does the suggested list just include specific apps, or can it include entire directories (like well-known Windows game directories) and/or allow wildcard matching?
+
+Implementation (estimated 40-65 points total)
+---------------------------------------------------
+
+This work is broken into four sections, with the bulk of code coming in the middle two sections. Those two sections should each be built behind their own separate feature flag, though we'd expose both parts together via Nimbus.
+
+### Prerequisites (21-36 points total)
+
+-   (3-5 points) Answer the open questions above
+-   (5-8 points) Create a well-structured file (for our own research, as well as one users can add to via GH pull requests on the client repo) w/ apps to exclude (and update the repo's readme to explain this new type of contribution). Add this doc to addons. Add tests to ensure the doc's format integrity is maintained. This format should be designed to allow new platforms (like websites) to be added without breaking anything.
```

Comment:
> Add this doc to addons.

Continuing the discussion from the Confluence.

How we are going to deploy this thorugh the addons framework needs to be better detailed and probably spun up into it's own task or group of tasks. Although in practice addons are just .rcc files served statically, they have a well defined structure in code, see: https://github.com/mozilla-mobile/mozilla-vpn-client/blob/main/docs/add-on.md#add-on-format.

Sure, .txt files can be added to .rcc files. Is that the best way to approach this though? Why .txt file and not something more easily digested from C++ code, akin to our featurelist or settinglist files? Even then, there are a few steps before that need to be figure out. Is this going to be a new type of addon? What type is that? What does the manifest.json for the new addon type look like? Is it going to be one addon per platform, one addon per suggested exclusion or one addon for everything? And so on.

I'd also question, if it is it absolutely crucial that this list have the dynamic properties of addons. We have been discussing for the past months the removal of a lot of the addons code. See VPN-5545 and most importantly VPN-5419. What are the advantages of serving this through addons? Is it worth the complexity trade-off over just having a file that is updated on each release?

Reply:
I think in light of your comments, and your reasonable request to want another set of tasks for this, we should consider this part dead, and fallback on the "just include the files in each release" plan. Will confirm with Santiago, and leave this open for 24 hours or so to ensure others can comment if they want to argue for serving this up dynamically.

Reply:
@albionx ^